crates/solver/src/liquidity.rs

```diff
@@ -212,8 +212,28 @@ impl LimitOrder {
     }
 }
 
+#[derive(Debug, Clone, PartialEq, Eq)]
+pub struct LimitOrderExecution {
+    /// The amount that the order `side` (`buy`, `sell`) should be filled by
+    /// this trade.
+    pub filled_amount: U256,
+    /// The fee associated with this order. Either determined by the solver for
+    /// partially fillable limit orders or determined by the protocol for
+    /// everything else.
+    pub executed_solver_fee: U256,
+}
+
+impl LimitOrderExecution {
+    pub fn new(filled: U256, solver_fee: U256) -> Self {
+        Self {
+            filled_amount: filled,
+            executed_solver_fee: solver_fee,
+        }
+    }
+}
+
 impl Settleable for LimitOrder {
-    type Execution = U256;
+    type Execution = LimitOrderExecution;
```

Comment:
I think this is the basic change that needs to happen and everything else follows from there.
@nlordell does that seem sensible to you?

Reply:
I think so as well.